internal/controller/apiextensions/composite/ready.go

```diff
@@ -41,7 +41,7 @@ const (
 	errFmtRequiresMatchConditions = "type %q requires a valid match condition"
 	errFmtRequiresMatchInteger    = "type %q requires a match integer"
 	errFmtUnknownCheck            = "unknown type %q"
-	errFmtRunCheck                = "cannot run readiness check at index %d"
+	errFmtRunCheck                = "cannot run readiness check for object %q of kind %q at index %d"
```

Comment:
```suggestion
	errFmtRunCheck                = "cannot run readiness check for %s/%s at index %d"
```

Would it more readable? e.g. _"cannot run readiness check for ProviderConfig/default at index 0"_